### PR TITLE
Added conditionals

### PR DESCRIPTION
**Pull request by strausr**
#### What does this PR solve?
Added conditional action

#### Final checklist
- [*] JSdoc - Add proper docstrings based on our PHP implementation
- [*] JSdoc - Hide private functions using @private
- [*] Implementation is aligned with PHP Reference(If different, please specify why)
- [*] Tests - Add proper tests to the added code

### CODE REVIEW

File: src/actions/condition/Condition.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      .map((val) => CONDITIONAL_OPERATORS[val] || val)
+      .join('_');
+
+
+    return resultExpression;
+  }
+}
+
+/**
+ * @param key
```
Comment (patrick-tolosa) [inline src/actions/condition/Condition.ts:41]: Please add the missing JSDoc (if we already know what they are from PHP 2.0)

If we don't know what they should be, please open an internal issue for future work to ensure the docstrings are added.

Comment (strausr) [reply src/actions/condition/Condition.ts]: updated with what is know and opened a ticket for unknowns

File: __TESTS__/unit/actions/Condition.test.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  });
+
+  it('Creates a cloudinaryURL with condition =', () => {
+    const url = new TransformableImage()
+      .setConfig(CONFIG_INSTANCE)
+      .ifCondition(Condition.fromString('ar = 1.0'))
+      .setPublicID('sample')
+      .toURL();
+
+    expect(url).toBe('http://res.cloudinary.com/demo/image/upload/if_ar_eq_1.0/sample');
```
Comment (patrick-tolosa) [inline __TESTS__/unit/actions/Condition.test.ts:28]: This transformation isn't valid, can we make sure our tests have valid and correct usages?
(this, for example, is missing an end-if)

Tests become a part of our informal documentation as to what's allowed and what's not

Comment (strausr) [reply __TESTS__/unit/actions/Condition.test.ts]: updated

**Pull request merged**